### PR TITLE
boost-ext-ut: Add version 1.1.9

### LINKED ISSUE #9663: [request] boost-ext-ut/1.1.9
### Package Details
  * Package Name/Version: boost-ext-ut/1.1.9
  * Changelog: https://github.com/boost-ext/ut/compare/v1.1.8...v1.1.9

The above mentioned version is newly released by the upstream project and not yet available as a recipe. Please add this version.
https://github.com/boost-ext/ut/issues/490
I requested this new version be released because 1.1.8 didn't have some fixes that were needed newer msvc versions. Like the ability to disable modules.

### PR DESCRIPTION
**Pull request by jwillikers**
This version fixes issues with newer MSVC compilers.
It also removes support for C++17.
Fixes #9663.

Specify library name and version:  **boost-ext-ut/1.1.9**

---

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by SSE4** [comment]
does it require C++ modules?

**Review by SSE4 — APPROVED** [review]

File: recipes/boost-ext-ut/all/conanfile.py [hunk 1]
```diff
@@ -30,13 +33,16 @@ def _minimum_compilers_version(self):
...
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def config_options(self):
+        if tools.Version(self.version) <= "1.1.8":
+            del self.options.disable_module
 
     def configure(self):
         if self.settings.compiler.get_safe("cppstd"):
             tools.check_min_cppstd(self, self._minimum_cpp_standard)
+        if tools.Version(self.version) <= "1.1.8" and self.settings.compiler in ["msvc", "Visual Studio"]:
```
Comment (uilianries) [inline recipes/boost-ext-ut/all/conanfile.py:43]: It should be moved to validate()

File: recipes/boost-ext-ut/all/conanfile.py [hunk 2]
```diff
@@ -52,17 +58,40 @@ def configure(self):
...
+        if self.settings.compiler in ["msvc", "Visual Studio"]:
+            if "disable_module" in self.options.values:
+                self.options.disable_module = True
+
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version])
-        os.rename("ut-" + self.version, self._source_subfolder)
-        tools.download("https://www.boost.org/LICENSE_1_0.txt", "LICENSE",
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True)
+        if tools.Version(self.version) <= "1.1.8":
```
Comment (uilianries) [inline recipes/boost-ext-ut/all/conanfile.py:67]: move license link and sha to conandata.yml

File: recipes/boost-ext-ut/all/conanfile.py (hunk 1 above)
Comment (jwillikers) [reply recipes/boost-ext-ut/all/conanfile.py:43]: Yes, of course. I've moved the checks from the `configure` method to the `validate` method.

File: recipes/boost-ext-ut/all/conanfile.py (hunk 2 above)
Comment (jwillikers) [reply recipes/boost-ext-ut/all/conanfile.py]: Done. The hooks didn't like the `license` keyword in the `conandata.yml` so I just used a patch to add the license.

**Review by uilianries — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**